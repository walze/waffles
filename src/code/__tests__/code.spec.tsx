import { render } from '@testing-library/react';

import { Code } from '../index';

const sizes = ['small', 'medium', 'large'] as const;

describe('Code', () => {
  it('renders a code element containing the text by default', () => {
    const { container } = render(<Code>npm run test -- -u</Code>);

    const codeElement = container.querySelector('code');

    expect(codeElement).toBeInTheDocument();
    expect(codeElement).toHaveTextContent('npm run test -- -u');
  });

  it('renders a pre element if it is passed via "as" prop', () => {
    const { container } = render(<Code as="pre">npm run test</Code>);

    const codeElement = container.querySelector('pre');

    expect(codeElement).toBeInTheDocument();
    expect(codeElement).toHaveTextContent('npm run test');
  });

  it('sets the data attribute on the code element', () => {
    const { getByTestId } = render(
      <Code data-testid="code-element">npm run test -- -u</Code>,
    );

    const codeElement = getByTestId('code-element');

    expect(codeElement).toBeInTheDocument();
  });

  describe('renders snapshot of', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { container } = render(<Code size={size}>test</Code>);

        const codeElement = container.querySelector('code');
        expect(codeElement).toMatchSnapshot();
      });
    });
  });

  describe('renders snapshot of inverted', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { container } = render(
          <Code size={size} inverted>
            test
          </Code>,
        );

        const codeElement = container.querySelector('code');
        expect(codeElement).toMatchSnapshot();
      });
    });
  });
});
