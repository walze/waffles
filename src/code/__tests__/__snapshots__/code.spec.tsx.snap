// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code renders snapshot of inverted size large 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  line-height: inherit;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: regular;
  font-size: 14px;
  color: #FFFFFF;
  background-color: rgba(255, 255, 255, 0.3);
}

<code
  class="emotion-0"
>
  test
</code>
`;

exports[`Code renders snapshot of inverted size medium 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  line-height: inherit;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: regular;
  font-size: 12px;
  color: #FFFFFF;
  background-color: rgba(255, 255, 255, 0.3);
}

<code
  class="emotion-0"
>
  test
</code>
`;

exports[`Code renders snapshot of inverted size small 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  line-height: inherit;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: regular;
  font-size: 10px;
  color: #FFFFFF;
  background-color: rgba(255, 255, 255, 0.3);
}

<code
  class="emotion-0"
>
  test
</code>
`;

exports[`Code renders snapshot of size large 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  line-height: inherit;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 14px;
  color: #05192D;
  background-color: #EFEBE4;
}

<code
  class="emotion-0"
>
  test
</code>
`;

exports[`Code renders snapshot of size medium 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  line-height: inherit;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 12px;
  color: #05192D;
  background-color: #EFEBE4;
}

<code
  class="emotion-0"
>
  test
</code>
`;

exports[`Code renders snapshot of size small 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  line-height: inherit;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 10px;
  color: #05192D;
  background-color: #EFEBE4;
}

<code
  class="emotion-0"
>
  test
</code>
`;
